math/krishnamurthy.c

```diff
@@ -0,0 +1,76 @@
+#include <stdio.h> // For I/O ops
+#include <assert.h> //For test cases
+
+/**
+SCRIPT DESCRIPTION:
+
+ * @description a krishnamurthy number is
+ * a number whose sum of the factorial of digits
+ * is equal to the number itself.
+
+ * @example 145 = 1! + 4! + 5!
+
+ * @reference https://www.geeksforgeeks.org/problems/krishnamurthy-number1323/1
+*/
+
+
+/**
+ * @description n! = n*(n-1)*(n-2)*...*1
+ * @brief calculates the factorial of a number
+ * @param n the number we want to calculate the factorial
+ * @return an integer representing the factorial of n
+ */
+int factorial(int n){
+    if (n<2){
+	return n;
+    }
+    return n*factorial(n-1);
+}
+
+/**
+ * @brief Calculates the sum of the factorial of each digit
+ * @param n the number to calculate the sum of the factorial of its digits
+ * @param sum the sum variable to be incremented 
+ * @returns an integer representing that sum
+ */
+int get_digit(int n, int sum){
```

Comment:
```suggestion
int get_digit(int n){
```
sum should be locally defined

Reply:
i even found a solution... i didn't even need sum!! thanks for your feedback, i learned something for free!! Thanks